tests/test_nonlocal_games/test_xor_game.py

```diff
@@ -7,178 +7,199 @@
+				[0, 0, 0, 1, 0],
+				[0, 0, 0, 0, 1],
+				[1, 0, 0, 0, 0],
+			]
+		)
+		odd_cycle = XORGame(prob_mat, pred_mat)
+		res = odd_cycle.quantum_value()
+		expected_res = 0.975528
+		self.assertEqual(np.isclose(res, expected_res), True)
+
+	def test_odd_cycle_game_classical_value(self):
+		"""Classical value for the odd-cycle game."""
+		prob_mat = np.array(
+			[
+				[0.1, 0.1, 0, 0, 0],
+				[0, 0.1, 0.1, 0, 0],
+				[0, 0, 0.1, 0.1, 0],
+				[0, 0, 0, 0.1, 0.1],
+				[0.1, 0, 0, 0, 0.1],
+			]
+		)
+
+		pred_mat = np.array(
+			[
+				[0, 1, 0, 0, 0],
+				[0, 0, 1, 0, 0],
+				[0, 0, 0, 1, 0],
+				[0, 0, 0, 0, 1],
+				[1, 0, 0, 0, 0],
+			]
+		)
+		odd_cycle = XORGame(prob_mat, pred_mat)
+		res = odd_cycle.classical_value()
+		expected_res = 0.9
+		self.assertEqual(np.isclose(res, expected_res), True)
+
+	def test_classical_parallel_rep(self):
+		"""Tests for classical value parallel reps."""
+		with self.assertRaises(ValueError):
+			prob_mat = np.array([[1 / 4, 1 / 4], [1 / 4, 1 / 4]])
+			pred_mat = np.array([[0, 0], [0, 1]])
+
+			game = XORGame(prob_mat, pred_mat, 2)
+			game.classical_value()
+
+	def test_negative_prob_mat(self):
+		"""Tests for invalid negative probability matrix."""
+		with self.assertRaises(ValueError):
+			prob_mat = np.array([[1 / 4, -1 / 4], [1 / 4, 1 / 4]])
+			pred_mat = np.array([[0, 0], [0, 1]])
+
+			game = XORGame(prob_mat, pred_mat)
+			game.quantum_value()
+
+	def test_invalid_prob_mat(self):
+		"""Tests for invalid probability matrix."""
+		with self.assertRaises(ValueError):
+			prob_mat = np.array([[1 / 4, 1], [1 / 4, 1 / 4]])
+			pred_mat = np.array([[0, 0], [0, 1]])
+
+			game = XORGame(prob_mat, pred_mat)
+			game.quantum_value()
+
+	def test_non_square_prob_mat(self):
+		"""Tests for invalid non-square probability matrix."""
+		with self.assertRaises(ValueError):
+			prob_mat = np.array([[1 / 4, 1 / 4, 1 / 4], [1 / 4, 1 / 4, 1 / 4]])
+			pred_mat = np.array([[0, 0], [0, 1]])
+
+			game = XORGame(prob_mat, pred_mat)
+			game.quantum_value()
+
+	def test_zero_prob_mat(self):
+		"""Tests for zero probability matrix."""
+		with self.assertRaises(ValueError):
+			prob_mat = np.array([[1 / 4, 0], [1 / 4, 0]])
+			pred_mat = np.array([[0, 0], [0, 1]])
+
+			game = XORGame(prob_mat, pred_mat)
+			game.quantum_value()
+
+	def test_chsh_game_nonsignaling_value(self):
+		"""Non-signaling value for the CHSH game."""
+		prob_mat = 1/4 * np.ones((2,2))
+		pred_mat = np.array([[0,0],[0,1]])
+
+		chsh = XORGame(prob_mat, pred_mat)
+		res = chsh.nonsignaling_value()
+		expected_res = 1
+		self.assertEqual(np.isclose(res, expected_res, atol=0.5), True)
+
+	def test_chsh_game_nonsignaling_value_rep_2(self):
+		"""Non-signaling value for the CHSH game for 2 reps."""
+		prob_mat = 1/4 * np.ones((2,2))
+		pred_mat = np.array([[0,0],[0,1]])
+
+		chsh = XORGame(prob_mat, pred_mat, 2)
+		res = chsh.nonsignaling_value()
+		expected_res = 1
+		self.assertEqual(np.isclose(res, expected_res, atol=0.5), True)
```

Comment:
This is a rather high `atol` value. I would suspect that we wouldn't need an `atol` param here at all, is that true?